3D/senior-csharp-test/Assets/Cache/FileCacheStorage.cs

```diff
@@ -29,136 +29,70 @@ public FileCacheEntry( string id, string version, string pathData, string lastUs
 
 	public FileCacheStorage( string path, string fileName, IFileSystem fileSystem, ISerializer serializer )
 	{
-		_cachePath = path;
-		_fileName = fileName;
-		_fileSystem = fileSystem;
-		_serializer = serializer;
-		InitCache();
-	}
-
-	public byte[] Get( string id )
-	{
-		FileCacheEntry ce;
+		if( string.IsNullOrEmpty( path ) )
+			throw new NullReferenceException( nameof( path ) );
 
-		if( !_cacheEntries.TryGetValue( id, out ce ) )
-			return null;
+		if( string.IsNullOrEmpty( fileName ) )
+			throw new NullReferenceException( nameof( fileName ) );
 
-		UpdateUseDate( id );
-		return _fileSystem.Read( ce.Data );
-	}
+		_cachePath = path;
+		_fileName = fileName;
 
-	public bool Has( string id )
-	{
-		if( _cacheEntries != null && _cacheEntries.ContainsKey( id ) )
-			return true;
+		_fileSystem = fileSystem ?? throw new NullReferenceException( nameof( fileSystem ) );
+		_serializer = serializer ?? throw new NullReferenceException( nameof( serializer ) );
 
-		return false;
+		InitCache();
 	}
 
-	public bool MatchesVersion( string id, string version )
+	private void InitCache()
 	{
-		try
+		if( !_fileSystem.DirExists( _cachePath ) )
 		{
-			if( _cacheEntries[ id ].Version == version )
-				return true;
+			_fileSystem.CreateDir( _cachePath );
+			return;
 		}
-		catch( Exception e )
+
+		if( TryLoadCacheEntries() )
 		{
-			Log.LogError( "Cache entry id = " + id + " not found : " + e.Message );
-			return false;
+			CheckCacheVersion();
+			ClearExpiredEntries();
 		}
-
-		return false;
-	}
-
-	public void Set( byte[] value, string id, string version = null )
```

Comment:
if `version` is null an exception will be thrown. having a `null` as a default value is not good

Reply:
Yes indeed, I think we don't want to specifiy a default value since it's not optional. Done